stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -334,32 +379,71 @@ def _search_for_move_domain(self, move):
             domain.append(("carrier_ids", "in", move.group_id.carrier_id.ids))
         else:
             domain.append(("carrier_ids", "=", False))
-        return domain
+        qty_uom_domain = expression.OR(
+            [
+                [("qty", "=", False)],
+                [
+                    ("uom_id.category_id", "=", move.product_uom_category_id.id),
+                ],
+            ]
+        )
+        return expression.AND([domain, qty_uom_domain])
 
-    def _is_valid_for_move(self, move):
-        self.ensure_one()
+    def _is_domain_valid_for_move(self, move):
         if not self.move_domain:
             return move
         domain = safe_eval(self.move_domain or "[]")
         if not domain:
             return move
         return move.filtered_domain(domain)
 
+    def _is_qty_valid_for_move(self, move):
+        if not self.qty:
+            return move
+        qty = self.uom_id._compute_quantity(self.qty, move.product_id.uom_id)
+        if qty <= move.product_qty:
```

Comment:
missing test on uom

Reply:
This should only be called after _search_for_move there is a Domain for uom

Reply:
But you are right it also makes sense to test it here